### PR TITLE
feat: build custom react-select component with Chakra [updated]

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
_Replaces #10542_

_Part of Implementation of new DS v1 #9548_

## Description

This PR update the usage of the `react-select` package.

- Bumps the `react-select` dep to latest which includes TypeScript support
- Creates a custom select component, including custom internal parts using Chakra components.
- Create a custom theming config `ReactSelect` to be sent to the internal parts
- Update implementation with related page components.
- Update onChange handler functions to match new TypeScript support.

### CODE REVIEW

File: src/components/ReactSelect/ReactSelect.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { useState } from "react"
+import { Meta, StoryObj } from "@storybook/react"
+
+import ReactSelect, { ReactSelectOnChange } from "."
+
+// TODO: Work with closed PR diff: https://github.com/ethereum/ethereum-org-website/pull/10542/files
```
Comment (TylerAPfledderer) [inline src/components/ReactSelect/ReactSelect.stories.tsx:6]: This storybook file is not complete. Should have a story displaying both style variants with the full logic combination including option groups and as combobox.

**Review by pettinarip — COMMENTED** [review]
Wow! this looks great @TylerAPfledderer!!! ❤️ 

Few things I saw while testing:

- Missaligned selects in the layer 2 page
image

- Group label color is black & white. Should be `primary.highContrast`
image
image

======

cc @nloureiro 👀 

Pages where we are using the select:
- https://deploy-preview-12065--ethereumorg.netlify.app/layer-2
- https://deploy-preview-12065--ethereumorg.netlify.app/get-eth
- https://deploy-preview-12065--ethereumorg.netlify.app/wallets/find-wallet
- https://deploy-preview-12065--ethereumorg.netlify.app/staking/solo

File: src/components/ReactSelect/ReactSelect.stories.tsx (hunk 1 above)
Comment (TylerAPfledderer) [reply src/components/ReactSelect/ReactSelect.stories.tsx:6]: @pettinarip along with your initial suggestions, I will ping you back when I have this story file finalized proper.

**Comment by nloureiro** [comment]
@TylerAPfledderer, great work, thank you!!! While testing on the browser, I realized that the color scheme was too strong and relied heavily on the primary color... Can you change it?

Sorry about this,  but sometimes, when on the browser and in the mix of the styles, we can find "design bugs".
Screen Shot 2024-02-14 11 49 49 AM

Link to the Figma file (I added a comment, too) https://www.figma.com/file/NrNxGjBL0Yl1PrNrOT8G2B/ethereum.org-Design-System?type=design&node-id=5303%3A58208&mode=design&t=gfwqptwT1wv8ZJCx-1

Thank you!

**Comment by TylerAPfledderer** [comment]
@pettinarip @nloureiro all above suggestions should be resolved, and have updated the storybook file.

**Comment by TylerAPfledderer** [comment]
I'm afk at the time of this comment, but I noticed a styling error for the flush variant. Will get that addressed soon.

**Comment by nloureiro** [comment]
> I'm afk at the time of this comment, but I noticed a styling error for the flush variant when checking the story book preview. Will get that addressed asap.

not sure what you mean by "flush variant".... do you mean the active color?

I was checking and still found this difference, not sure if this is what you mean
Screen Shot 2024-02-15 09 38 11 AM

**Comment by TylerAPfledderer** [comment]
> > I'm afk at the time of this comment, but I noticed a styling error for the flush variant when checking the story book preview. Will get that addressed asap.
> 
> not sure what you mean by "flush variant".... do you mean the active color?
> 
> I was checking and still found this difference, not sure if this is what you mean Screen Shot 2024-02-15 09 38 11 AM

I'm referring to the `flushed` versus `outline` variant you have labelled in the DS. I made an update to the border styling which caused the variants to look the same.

This should now be resolved for a proper review.

**Comment by TylerAPfledderer** [comment]
@nloureiro you can immediately view this component through this [published storybook preview](https://63b7ea99632763723c7f4d6b-nqqjiytipk.chromatic.com/?path=/story/atoms-form-dropdown--dropdown)

**Review by nloureiro — APPROVED** [review]

**Comment by nhsz** [comment]
@TylerAPfledderer @pettinarip is this ready to be merged? I need to use a `Select` component and would be very handy, thanks!

**Comment by TylerAPfledderer** [comment]
> @TylerAPfledderer @pettinarip is this ready to be merged? I need to use a `Select` component and would be very handy, thanks!

@nhsz I think this is good to go for the most part. What's left is removing the [original select and it's story](https://github.com/ethereum/ethereum-org-website/tree/dev/src%2Fcomponents%2FSelect)

**Review by pettinarip — APPROVED** [review]
🚀

**Pull request merged**